MetaMask/metamask-extension
# Replace deprecated components for <ConfirmationNetworkSwitch/>. Refactor code to abstract styles to .scss and reduce repeating code.

## Explanation

1. Part of #17670
2. Part of #19526
3. Part of #20459
4. Centered network icons within their containers.
5. Abstracted imported style props in `<ConfirmationNetworkSwitch/>` to css classes.
6. Abstracted repeating icon jsx to `<NetworkIcon/>`.
7. Reduced lines of code in `confirmation-network-switch.js` by ~30%.

### Before

<img width="1124" alt="Screenshot 2023-09-29 at 2 12 53 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/6d57649d-6428-4b12-b3b3-2b085641a139">


https://github.com/MetaMask/metamask-extension/assets/8112138/9da94cbf-57d3-4f5b-bf0a-9135b6b67366



### After

<img width="1075" alt="Screenshot 2023-09-29 at 2 12 22 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/0cdceefd-b255-4f44-9c0b-f0d5343248e8">


https://github.com/MetaMask/metamask-extension/assets/8112138/a685c900-0a7e-4ce6-be2d-5cebf3e10ab4



## Manual Testing Steps

1. Trigger the prompt to switch blockchain networks. For example:
    a) Set MM network to Ethereum.
    b) Connect MM to [Chainlist](https://chainlist.org/).
    c) Change networks to Polygon from within Chainlist.
3. Observe the network icons are centered within their containers, including unknown networks.
4. Observe all other text and UI elements are unchanged.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
